### ISSUE TITLE
statistics callback

### ISSUE THREAD

**Issue by ctramnitz** [edenhill/kafkacat#79]
Is there a way to register the stats callback in librdkafka to get runtime statistics (transmitted messages) output by kafkacat? If not is there another way to get any metrics out of kafkacat? We would like to put this into graphite for performance monitoring...

**Comment by edenhill** [comment]
Currently no.

How would you like these stats to be exposed?
stdout only works for the producer, stderr is used by error logs so that might cause some mixups.

What I've done in rdkafka_performance is add a `-Y <cmd>` flag which executes `<cmd>` and pipes the stats to it (e.g., analogue to `generate_stats.. | <cmd>`.
Would that work for you?

**Comment by ctramnitz** [comment]
Yes that would be perfect using `-Y <cmd>`. The format should be parsable. We would use `cmd` to pipe it into graphite then.  How about the stats interval in such a scenario? Would it have an own interval or would be pass through `statistics.interval.ms` somehow?

**Comment by edenhill** [comment]
The format will be one JSON object (the stats object in its entirety) per line.
statistics.interval.ms will be configured through the usual means (`-X statistics.interval.ms=..`)

**Comment by fsaintjacques** [comment]
why not just `-Y <file>` and then one can use bash's `<()` construct to emulate piping...

**Comment by edenhill** [comment]
That would make it depend on bash to get the same functionality.
